tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -243,9 +243,9 @@ public bool AsBool(bool defaultValue, Func<bool, bool>? validator)
         // ****************
         // Dictionary accessors
         // ****************
-        public IDictionary<string, string>? AsDictionary() => AsDictionary(allowOptionalMappings: false);
+        public IDictionary<string, string>? AsDictionary(Func<IDictionary<string, string>>? getDefaultValue = null) => AsDictionary(allowOptionalMappings: false, getDefaultValue: getDefaultValue);
 
-        public IDictionary<string, string>? AsDictionary(bool allowOptionalMappings)
+        public IDictionary<string, string>? AsDictionary(bool allowOptionalMappings, Func<IDictionary<string, string>>? getDefaultValue = null)
```

Comment:
```suggestion
        [return: NotNullIfNotNull(nameof(getDefaultValue))]
        public IDictionary<string, string>? AsDictionary(bool allowOptionalMappings, Func<IDictionary<string, string>>? getDefaultValue = null)
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5126/commits/7d4b6a84d089ba85674e50047ce60bc18a403cfc